projects/cds/src/cms-components/merchandising/carousel/merchandising-carousel/merchandising-carousel.component.ts

```diff
@@ -0,0 +1,51 @@
+import { ChangeDetectionStrategy, Component } from '@angular/core';
+import { Product } from '@spartacus/core';
+import { CmsComponentData } from '@spartacus/storefront';
+import { Observable, of } from 'rxjs';
+import { distinctUntilChanged, filter, map, switchMap } from 'rxjs/operators';
+import { CmsMerchandisingCarouselComponent } from '../../../../cds-models/cms.model';
+import { StrategyConnector } from './../../../../merchandising/connectors/strategy/strategy.connector';
+
+@Component({
+  selector: 'cx-merchandising-carousel',
+  templateUrl: './merchandising-carousel.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class MerchandisingCarouselComponent {
+  private componentData$: Observable<
+    CmsMerchandisingCarouselComponent
+  > = this.componentData.data$.pipe(filter(Boolean));
+
+  title$: Observable<string> = this.componentData$.pipe(
+    map(data => data.title)
+  );
+
+  size$: Observable<string> = this.componentData$.pipe(
```

Comment:
I know we need size$, backgroundColor$, and textColor$ as merch carousel supports these additional properties, but we shouldn't include it in this PR until make changes to the current carousel implementation to support these changes.

Reply:
I've removed them for now, however, size relates more to how many products are returned from the strategy service. We have another ticket to include that functionality in the next sprint. One question though, if we move this stuff from the component, it still shows in the configuration of the CMS component in smartedit....is that ok?

Reply:
I think it would be best to include these 'configs' in the next ticket as well as it is not used as of yet. Like I mentioned, I know it is necessary since these config exists for the merch carousel and used in smartedit, but the current carousel does not support these inputs as of yet.

Reply:
ok, I've done that, cheers

Reply:
I'd argue about using `size$` going forward. A single static size property doesn't take care of different breakpoints (i.e 4 items on desktop vs 2 items on tablet vs 1 item on mobile). It also doesn't support components being (re)used in multiple grid system (i.e. using a carousel in a 3 100% width grid vs 50% width grid). 

This is where the `itemWidth` kicks in. The intend of the `itemWidth` is that it will calculate the number of items (size) for the available size vs `itemWidth`. 

If you don't agree, I'm ready to learn about the requirements.

Reply:
Regarding `backgroundColor$` and `textColor$`, they should not be implemented in the `CarouselComponent`. They can be implemented in the `MerchandisingCarouselComponent`, as the `MerchandisingCarouselComponent` can manage those properties itself, without being bothered by the `CarouselComponent`.

Reply:
So, how does the existing product carousel limit the number of products returned for display. i.e. a merchandiser might want to be able to have 20 products in one carousel, but 5 in another, or 100 in another. The size relates to the number of items returned from the API endpoint. Currently this has a default limit of 16, but surely this should be configurable, no? This size property doesn't relate to the number of products to display, as such....if that makes sense

Reply:
I'm happy to implement the background colour/text colour in our own component only